tests/test_flatten.py

```diff
@@ -8,18 +13,29 @@
         "contextInfo": {"context1": "info1", "context2": "info2"},
     }
 ]
+PD_DATA = DataFrame(DICT_DATA)
+GEOMETRY = [Point(4, 2)]
+GEOPD_DATA = GeoDataFrame(DICT_DATA, crs="EPSG:4326", geometry=GEOMETRY)
 
 
 def teste_flatten_wrong_nested_columns_value_error():
     with raises(NestedColumnError):
         flatten(DICT_DATA, nested_columns=["wrong"])
 
 
-def teste_flatten_with_emptylist():
+def teste_flatten_with_empty_list():
     assert flatten([]) == []
 
 
-# test the flatten function with a dictionary mocking it to assert _flatten_dict function is being called
+def teste_flatten_with_empty_data_frame():
+    mock_flatten_df = Mock()
+
+    with patch("crossfire._flatten_df", mock_flatten_df):
```

Comment:
Please, review the docs for `patch` (or other use cases in this project)  because this test does not do what you think it is doing. It mocks, but the mock you assert is not the patched one (it is the one you created manually on line 31).